state/state.go

```diff
@@ -1118,6 +1118,47 @@ var (
 	errLocalApplicationExists          = errors.Errorf("application already exists")
 )
 
+type AddCloudServiceArgs struct {
+	// Id will be the application Name if it's a part of application,
+	// and will be controller UUID for k8s a controller(controller does Not have an application).
+	Id         string
+	ProviderId string
+	Addresses  []network.Address
+}
+
+// AddCloudService creates a cloud service.
+func (st *State) AddCloudService(args AddCloudServiceArgs) (_ *CloudService, err error) {
```

Comment:
This needs to be called "SaveCloudService" since the saveServiceOps() below does either an insert or update as needed. AddCloudService() would be a lie sometimes